integration/test.sh

```diff
@@ -39,71 +39,74 @@ cf delete-buildpack $BUILDPACK_NAME -f
 # Create buildpack
 cf create-buildpack $BUILDPACK_NAME $BUILDPACK_VERSION_1_ZIP 100 --enable
 
-pushd app
-# deploy the app
-make deploy
-
-# Run buildpack notify app
-pushd ../../
-go build && ./cg-buildpack-notify -notify > log.txt
-## show the log.
-echo "Showing run log.."
-cat log.txt
-## check log
-echo "Searching for text.."
-grep "Sent e-mail to" log.txt
-RESULT=$?
-if [ $RESULT -eq 0 ]; then
-  echo "base case. should not find any users."
-  RET=1
-else
-  echo "base case success. shouldn't find any users."
-fi
-popd
-
-# update the buildpack
-cf update-buildpack $BUILDPACK_NAME -p $BUILDPACK_VERSION_2_ZIP -i 100 --enable
-
-# Run buildpack notify app
-pushd ../../
-go build && ./cg-buildpack-notify -notify > log.txt
-## show the log.
-echo "Showing run log.."
-cat log.txt
-## check log
-echo "Searching for text.."
-grep "Sent e-mail to $TEST_USER" log.txt
-RESULT=$?
-if [ $RESULT -eq 0 ]; then
-  echo "success"
-else
-  echo "didn't send e-mail to user $TEST_USER"
-  RET=1
-fi
-popd
-
-# deploy the app again
-cf restage dummy-app
-
-# Run buildpack notify app
-pushd ../../
-go build && ./cg-buildpack-notify -notify > log.txt
-## show the log.
-echo "Showing run log.."
-cat log.txt
-## check log
-echo "Searching for text.."
-grep "Sent e-mail to" log.txt
-RESULT=$?
-if [ $RESULT -eq 0 ]; then
-  echo "after update case. should not find any users"
-  RET=1
-else
-  echo "after update case success. shouldn't find any users."
-fi
-popd
-
-popd
+pushd app || exit
```

Comment:
the `pushd || exit` pattern is an improvement, but I wonder if it should be `pushd || exit 1`?